framework/include/bcs/PostprocessorDirichletBC.h

```diff
@@ -1,42 +0,0 @@
-/****************************************************************/
-/*               DO NOT MODIFY THIS HEADER                      */
-/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
-/*                                                              */
-/*           (c) 2010 Battelle Energy Alliance, LLC             */
-/*                   ALL RIGHTS RESERVED                        */
-/*                                                              */
-/*          Prepared by Battelle Energy Alliance, LLC           */
-/*            Under Contract No. DE-AC07-05ID14517              */
-/*            With the U. S. Department of Energy               */
-/*                                                              */
-/*            See COPYRIGHT for full restrictions               */
-/****************************************************************/
-
-#ifndef POSTPROCESSORDIRICHLETBC_H
-#define POSTPROCESSORDIRICHLETBC_H
-
-#include "NodalBC.h"
-
-class PostprocessorDirichletBC;
-
-template<>
-InputParameters validParams<PostprocessorDirichletBC>();
-
-/**
- * Boundary condition of a Dirichlet type
- *
- * Sets the value at the node to the value of a Postprocessor
- */
-class PostprocessorDirichletBC : public NodalBC
```

Comment:
This is NOT unnecessary!  Remember one of the basic tenets of software engineering: A class should one (and only one!) reason to exist!

There is nothing wrong with having very particular classes that do their one job well... and don't have ambiguous types for parameters...